xarray/core/merge.py

```diff
@@ -93,30 +95,40 @@ def unique_variable(name, variables, compat="broadcast_equals"):
     MergeError: if any of the variables are not equal.
     """  # noqa
     out = variables[0]
-    if len(variables) > 1:
-        combine_method = None
 
-        if compat == "minimal":
-            compat = "broadcast_equals"
+    if len(variables) == 1 or compat == "override":
+        return out
+
+    combine_method = None
+
+    if compat == "minimal":
+        compat = "broadcast_equals"
 
-        if compat == "broadcast_equals":
-            dim_lengths = broadcast_dimension_size(variables)
-            out = out.set_dims(dim_lengths)
+    if compat == "broadcast_equals":
+        dim_lengths = broadcast_dimension_size(variables)
+        out = out.set_dims(dim_lengths)
 
-        if compat == "no_conflicts":
-            combine_method = "fillna"
+    if compat == "no_conflicts":
+        combine_method = "fillna"
 
+    if equals is None:
+        out = out.compute()
+        equals_list = []
         for var in variables[1:]:
-            if not getattr(out, compat)(var):
-                raise MergeError(
-                    "conflicting values for variable %r on "
-                    "objects to be combined:\n"
-                    "first value: %r\nsecond value: %r" % (name, out, var)
-                )
-            if combine_method:
-                # TODO: add preservation of attrs into fillna
-                out = getattr(out, combine_method)(var)
-                out.attrs = var.attrs
+            equals_list.append(getattr(out, compat)(var.compute()))
```

Comment:
Is there a reason why you're now explicitly computing everything ahead of time? In the previous implementation, at least we would not complete everything if there was a failure in comparison between the first two variables.

It seems like the ideal thing to do (in the case of dask) might be to build up a single computation graph and call compute once? Certainly we can save that for later, though.

Reply:
> Is there a reason why you're now explicitly computing everything ahead of time?

This was a mistake. I've fixed it.

Reply:
I don't see any changes to this logic?

We do have inconsistent behavior here currently (on master), which your PR would resolve:
- Inside `merge()`, variables are computed *after* doing the equality comparison. 
- Inside `concat()`, variables are computed *before* doing the equality comparison.

We could certainly change this for consistency, but I'm not 100% sure which is the better option.

Reply:
Ah now I understand, we want to check attrs, shapes etc. before comparing values. How about

``` python
    if equals is None:
        out = out.compute()
        for var in variables[1:]:
            equals = getattr(out, compat)(var)
            if not equals:
                break
```

This still passes the dask test (`test_concat_loads_variables` in `test_dask.py`) but `out` is still being computed before doing any checks. Not sure how to get around this.

Reply:
This looks pretty reasonable to me. @crusaderky any thoughts?

Reply:
I guess the proper solution (for a different PR) is to have a function that loops through a list of variables and compares everything but values; and then if needed, loops again and compares values.